src/zino/trapobservers/logged_traps.py

```diff
@@ -0,0 +1,17 @@
+"""This module implements custom logging of certain SNMP traps, but takes no other action"""
+import logging
+from typing import Optional
+
+from zino.trapd import TrapMessage, TrapObserver
+
+_logger = logging.getLogger(__name__)
+
+
+class RestartTrapLogger(TrapObserver):
+    WANTED_TRAPS = {
+        ("SNMPv2-MIB", "coldStart"),
+        ("SNMPv2-MIB", "warmStart"),
+    }
+
+    def handle_trap(self, trap: TrapMessage) -> Optional[bool]:
+        _logger.info("%s: %s", trap.agent.device.name, trap.name)
```

Comment:
Shouldn't this also return `False`? Since that is the only thing that is done with these traps and therefore we can stop processing this trap afterwards?

Reply:
The return value is implicitly False.

The annotated return type is `Optional[bool]`.  If there is no return statement, `None` is Python's default return value.  `bool(None) == False`.

Anyway, `explicit is better than implicit`, so we could still add an explicit return :)